### PR TITLE
Fix root folder reference

### PR DESCRIPTION
**Pull request by neel-astro**
Changes:

- Fix a case when root-folder is not specified, and deploy-action would cd into the home folder, and fail to fetch updated files as a result
- Fix a case where the `image-and-dags` deploy type would skip deploy when a dag file is updated
- Fix the e2e test case which was relying on actual `git` still, as I forgot to add mock to it, because of which the outcome of the test varied based on the changes in the commit 

Because we can't test this edge case as part of the e2e test suite, I had to test it in a separate repo: https://github.com/neel-astro/example-repo-update/actions/workflows/push-to-main.yaml?query=branch%3Amain

### CODE REVIEW

File: .github/workflows/tests.yaml [hunk 1]
```diff
@@ -377,6 +377,13 @@ jobs:
       - name: Set CLI context
         run: astro context switch ${{ steps.get-astro-env-info.outputs.astronomer_host }}
 
+      - name: Mock git commands for DAG Deploy
+        run: |
+
+          # set to no deploy git mock, to simulate no interested file change scenario
+          mv e2e-setup/mocks/no-deploy-git.sh /usr/local/bin/git
```
Comment (neel-astro) [inline .github/workflows/tests.yaml:384]: fixing a case that I found after merging my previous PR, as for the no-deploy case we want to simulate a change in a file outside the root folder. This is

File: action.yaml [hunk 2]
```diff
@@ -7,7 +7,7 @@ branding:
 inputs:
   root-folder:
     required: false
-    default: ""
+    default: "./"
```
Comment (jeremybeard) [inline action.yaml:10]: Can we leave the default empty and then only do the `cd` if it is set to something non-empty?

File: action.yaml [hunk 3]
```diff
@@ -342,24 +347,34 @@ runs:
         cd ${{ inputs.root-folder }}
         branch=$(echo "${GITHUB_REF#refs/heads/}")
         echo "Branch pushed to: $branch"
+        pwd
```
Comment (jeremybeard) [inline action.yaml:350]: Was this just for debugging?

File: action.yaml [hunk 4]
```diff
@@ -342,24 +347,34 @@ runs:
...
-          if [[ $file =~ ^"${{ inputs.root-folder }}".* ]]; then
+          if [[ $file =~ ^"$root_folder".* ]]; then
             echo $file is part of the input root folder
             SKIP_IMAGE_OR_DAGS_DEPLOY=false
-            if [[ $file == *"dags/"* && ${{ inputs.deploy-type }} != 'image-and-dags' ]]; then
-              echo $file is part of dags folder, triggering a DAG-only deploy
-              DAGS_ONLY_DEPLOY=true
+            if [[ $file != *"dags/"* ]]; then
+              echo $file is not part of dags folder
+              DAGS_ONLY_DEPLOY=false
```
Comment (jeremybeard) [inline action.yaml:377]: Good catch...

File: action.yaml [hunk 5]
```diff
@@ -308,8 +308,13 @@ runs:
...
         DBT_DEPLOY=false
 
+        root_folder="${{ inputs.root-folder }}"
+        if [[ $root_folder == "./" ]]; then
+          root_folder=""
+        fi
+
         for file in $files; do
-          if [[ $file =~ ^"${{ inputs.root-folder }}".* ]]; then
+          if [[ $file =~ ^"$root_folder".* ]]; then
```
Comment (jeremybeard) [inline action.yaml:317]: Do we also need some kind of validation that the root folder does not start with a forward slash, which presumably would not match with any `file`?

File: .github/workflows/tests.yaml
Comment (jeremybeard) [inline .github/workflows/tests.yaml:1]: Should we add a test where we don't set any optional inputs on the action? To exercise the case where all the defaults are used.

**Review by jeremybeard — APPROVED** [review]
Approving ahead of some comments to address.

File: action.yaml (hunk 3 above)
Comment (neel-astro) [reply action.yaml:350]: yeah 😅

File: .github/workflows/tests.yaml
Comment (neel-astro) [reply .github/workflows/tests.yaml:1]: added a test for the default case, had to catch "astro deploy" and tweak it so that we could still use the `e2e-setup/astro-project` for the tests, instead of having an astro project at the base level

File: action.yaml (hunk 5 above)
Comment (neel-astro) [reply action.yaml]: issue to track the change: https://github.com/astronomer/deploy-action/issues/85

Comment (sphvn) [reply action.yaml]: Currently we run `astronomer/deploy-action@v0.7` prior to `astro deployment variable update` in CI. If there are no changes, this step fails. Will set the `root-folder` to see if that gets us moving forward. Ideally these changes could be versioned, if possible, though.

Comment (neel-astro) [reply action.yaml]: yup, since this is a fix, we are planning to release a `v0.7.1` with these PR changes
